### PR TITLE
Test the suspenders:production:email generator

### PR DESCRIPTION
**Pull request by mike-burns**
This is the first isolated test for a generator in this project. It
introduces a few new test helpers:

- `#with_app` takes a block and runs the block after ensuring a fresh
app.
- `#generate` runs the specified generator.
- `#match_contents` is a matcher that takes a regexp as an argument and
applies in the context for a filename expectation. It reads the file
within the dummy app and returns whether the given regexp matches the
contents of the file.
- `#rails_new` generates a Rails app.
- `#system_rails_bin` is the Rails binary.
- `#project_rails_bin` is the `bin/rails` within an app.

This test runs in isolation in approximately five seconds on my
computer. This is both slow and one of the fastest tests in the repo.

---

While here, remove two race conditions because filesystem race conditions irk me.

### CODE REVIEW

File: spec/support/suspenders.rb [hunk 1]
```diff
@@ -29,6 +29,51 @@ def run_suspenders(arguments = nil)
...
+        end
+      end
+    end
+  end
+
+  def generate(generator)
+    Dir.chdir(project_path) do
+      Bundler.with_clean_env do
+        with_revision_for_honeybadger do
+          `bin/spring stop`
```
Comment (mike-burns) [inline spec/support/suspenders.rb]: This is based on lessons learned in 3cf23ed37a94a890f7678a8ec5cd072385f93bb4.

File: spec/support/suspenders.rb [hunk 2]
```diff
@@ -40,23 +85,23 @@ def suspenders_help_command
   end
 
   def setup_app_dependencies
-    if File.exist?(project_path)
-      Dir.chdir(project_path) do
-        Bundler.with_clean_env do
-          `bundle check || bundle install`
-        end
+    Dir.chdir(project_path) do
+      Bundler.with_clean_env do
```
Comment (georgebrock) [inline spec/support/suspenders.rb]: This pattern of `Dir.chdir(project_path) { Bundler.with_clean_env { … } }` seems to be repeated quite a lot. How about a `#run_in_project` method?

File: spec/support/suspenders.rb [hunk 3]
```diff
@@ -29,6 +29,52 @@ def run_suspenders(arguments = nil)
     end
   end
 
+  def with_app(&block)
+    drop_dummy_database
+    remove_project_directory
+    rails_new
+    setup_app_dependencies
+
+    block.call
```
[the remark this answers is not in the document]
Comment (georgebrock) [reply spec/support/suspenders.rb]: I agree with the Hound on this one.

File: spec/support/match_contents_matcher.rb [hunk 4]
```diff
@@ -0,0 +1,6 @@
+RSpec::Matchers.define :match_contents do |regexp|
+  match do |filename|
+    contents = IO.read(File.join(project_path, filename))
```
Comment (seanpdoyle) [inline spec/support/match_contents_matcher.rb]: Would [`Pathname(project_path, filename).read`](https://ruby-doc.org/stdlib-2.1.3/libdoc/pathname/rdoc/Pathname.html#method-i-read) yield the same value for `contents`?

Comment (seanpdoyle) [reply spec/support/match_contents_matcher.rb]: Ignore that, `Pathname` and `Pathname.new` don't take variable arguments, they both expect a single argument string.

Comment (mike-burns) [reply spec/support/match_contents_matcher.rb]: I couldn't find a `Pathname` constructor that takes multiple arguments, unfortunately -- so it'd still be `Pathname(File.join(project_path, filename)).read`. Dunno if that reads any more nicely.

Comment (mike-burns) [reply spec/support/match_contents_matcher.rb]: Ah you just said that. Yup!

**Pull request merged**